src/incentives/AffinePass.sol

```diff
@@ -0,0 +1,182 @@
+        bytes32 node = keccak256(abi.encodePacked(user));
+        return MerkleProof.verify(proof, merkleRoot, node);
+    }
+
+    function toggleWhitelistSale() public onlyOwner {
+        whitelistSaleIsActive = !whitelistSaleIsActive;
+    }
+
+    function togglePublicSale() public onlyOwner {
+        saleIsActive = !saleIsActive;
+        whitelistSaleIsActive = false;
+    }
+
+    function mintReserve(uint256 amount) public onlyOwner {
+        require(totalSupply() + amount <= MAX_SUPPLY, "Exceeds max supply");
+        require(reserveTokens + amount <= MAX_RESERVE_TOKENS, "Exceeds max reserve supply");
+        reserveTokens += amount;
+        for (uint256 i = 0; i < amount; i++) {
+            uint256 tokenId = _tokenIdCounter.current();
+            _tokenIdCounter.increment();
+            _safeMint(owner(), tokenId);
+        }
+    }
+
+    function hasRemainingSupply() public view returns (bool) {
+        uint256 currentSupply = totalSupply() - reserveTokens;
+        return currentSupply < MAXMINTABLESUPPLY;
+    }
+
+    function mintDrop(address[] memory recipients, uint256[] memory quantities) public onlyOwner {
+        require(recipients.length == quantities.length, "Recipients and quantities length mismatch");
+        for (uint256 i = 0; i < recipients.length; i++) {
+            for (uint256 j = 0; j < quantities[i]; j++) {
+                require(totalSupply() + quantities[i] <= MAX_SUPPLY, "Exceeds max supply");
+                uint256 tokenId = _tokenIdCounter.current();
+                _tokenIdCounter.increment();
+                _safeMint(recipients[i], tokenId);
+            }
+        }
+    }
+    
+    function mintWhitelist(bytes32[] memory proof) public payable {
+        require(
+                (whitelistSaleIsActive && isWhitelisted(_msgSender(), proof)),
+            "Sale paused or not whitelisted"
+        );
+        require(hasRemainingSupply(), "Exceeds max supply");
+        require(
+            _whitelistMinted[_msgSender()] + 1 <= MAX_WHITELIST_MINT,
+            "Exceeds max WL mint"
+        );
+
+        _whitelistMinted[_msgSender()] += 1;
+
+        uint256 tokenId = _tokenIdCounter.current();
+        _tokenIdCounter.increment();
+        _safeMint(_msgSender(), tokenId);
+    }
+
+    function mint() public payable {
+        require(saleIsActive, "Sale is not active");
+        require(hasRemainingSupply(), "Exceeds max supply");
+        require(
+            _minted[_msgSender()] + 1 <= MAX_PUBLIC_MINT,
+            "Exceeds max public mint"
+        );
+
+        _minted[_msgSender()] += 1;
+
+        uint256 tokenId = _tokenIdCounter.current();
+        _tokenIdCounter.increment();
+        _safeMint(_msgSender(), tokenId);
+
+    }
+
+    function withdraw() public onlyOwner {
+        uint256 balance = address(this).balance;
+        payable(owner()).transfer(balance);
+    }
+
+    function _baseURI() internal view virtual override returns (string memory) {
```

Comment:
This function is unused, can you delete it?

Reply:
It's an internal override, it's used by ERC721's tokenURI public read function